webpack.config.js

```diff
@@ -75,10 +77,60 @@ const externals = {
 	};
 } );
 
+/**
+ * Webpack plugin for handling specific template tags in Webpack configuration
+ * values like those supported in the base Webpack functionality (e.g. `name`).
+ *
+ * @see webpack.TemplatedPathPlugin
+ */
+class CustomTemplatedPathPlugin {
+	/**
+	 * CustomTemplatedPathPlugin constructor. Initializes handlers as a tuple
+	 * set of RegExp, handler, where the regular expression is used in matching
+	 * a Webpack asset path.
+	 *
+	 * @param {Object.<string,Function>} handlers Object keyed by tag to match,
+	 *                                            with function value returning
+	 *                                            replacement string.
+	 *
+	 * @return {void}
+	 */
+	constructor( handlers ) {
+		this.handlers = reduce( handlers, ( result, handler, key ) => {
+			const regexp = new RegExp( `\\[${ escapeRegExp( key ) }\\]`, 'gi' );
+			return [ ...result, [ regexp, handler ] ];
+		}, [] );
+	}
+
+	/**
+	 * Webpack plugin application logic.
+	 *
+	 * @param {Object} compiler Webpack compiler
+	 *
+	 * @return {void}
+	 */
+	apply( compiler ) {
+		compiler.plugin( 'compilation', ( compilation ) => {
+			compilation.mainTemplate.plugin( 'asset-path', ( path, data ) => {
+				for ( let i = 0; i < this.handlers.length; i++ ) {
+					const [ regexp, handler ] = this.handlers[ i ];
+					if ( regexp.test( path ) ) {
+						return path.replace( regexp, handler( path, data ) );
+					}
+				}
+
+				return path;
+			} );
+		} );
+	}
+}
+
 const config = {
 	entry: Object.assign(
-		entryPointNames.reduce( ( memo, entryPointName ) => {
-			memo[ entryPointName ] = `./${ entryPointName }/index.js`;
+		entryPointNames.reduce( ( memo, entryPoint ) => {
+			entryPoint = castArray( entryPoint );
+			const [ name, path = name ] = entryPoint;
```

Comment:
Just wanted to double check if I understand how destructuring works here:
* if there is a `name` and `path` provided they are set
* if the `path` is not set, we set its value as `name`

Reply:
>Just wanted to double check if I understand how destructuring works here:

This is correct. I've added a clarifying comment in d3a43bf.

Reply:
Thanks, you don't always see defaults set here, using another param. Quite nice that is is possible 👍